catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -74,15 +87,22 @@ public RemoteSolrCatalogProvider(
       SolrClient client,
       SolrFilterDelegateFactory solrFilterDelegateFactory,
       @Nullable DynamicSchemaResolver resolver) {
-    this.provider =
-        new SolrCatalogProvider(
-            client,
-            filterAdapter,
-            solrFilterDelegateFactory,
-            (resolver == null) ? new DynamicSchemaResolver() : resolver);
+    this.filterAdapter = filterAdapter;
+    this.client = client;
+    this.solrFilterDelegateFactory = solrFilterDelegateFactory;
+    this.resolver = resolver;
+    init();
     provider.maskId(getId());
   }
 
+  protected void init() {
+    if (resolver == null) {
+      resolver = new DynamicSchemaResolver(anyTextAttributes, Collections.emptyList());
+    }
+    this.provider =
```

Comment:
❗️ This should also be done in the ctor to clarify that the `provider` attribute should never be `null` in this class. Plus it would allow it to remain `final` making it clear it doesn't change and making thus avoiding any potential multi-threading issue around it.